src/compiler/checker.ts

```diff
@@ -8822,6 +8959,177 @@ module ts {
             }
         }
 
+        function checkNonThenableType(type: Type, location?: Node, message?: DiagnosticMessage) {
+            if (!allConstituentTypesHaveKind(type, TypeFlags.Any) && isTypeAssignableTo(type, getGlobalThenableType())) {

```

Comment:
I plan to change mine as well, so that it does not call allConstituentTypesHaveKind